### PR TITLE
OJ-30236 download teams

### PR DESCRIPTION
**Pull request by thingscouldbeworse**

### CODE REVIEW

**Review by gavinpitt-jf — APPROVED** [review]
Seems safe 👍 

If you want to hide the error logs from Agent users, but make them show up in the .log file, you can set the log messages to be at the `DEBUG` level. Totally up to you on if you think that's a better UX though

File: jf_agent/jf_jira/jira_download.py [hunk 1]
```diff
@@ -859,6 +859,26 @@ def download_customfieldoptions(jira_connection, project_ids):
...
+    logger.info('downloading jira teams... [!n]')
+    server = jira_connection._options['server']
+    try:
+        teams_url = jira_connection.JIRA_BASE_URL.format(
+            server=server,
+            rest_path='teams-api',
+            rest_api_version='1.0',
+            path='team'
+        )
+        teams = jira_connection._get_json('team', base=teams_url)
```
Comment (jruel4) [inline jf_agent/jf_jira/jira_download.py:874]: Nit: Use `retry_for_429s` here to handle rate-limiting

File: jf_agent/jf_jira/__init__.py [hunk 2]
```diff
@@ -302,11 +303,18 @@ def download_and_write_issues_needing_re_download():
...
         )
+        try:
+            write_file(
+                config.outdir,
+                'jira_teams',
+                config.compress_output_files,
+                download_teams(jira_connection)
+            )
+        except Exception as e:
+            logging_helper.log_standard_error(logging.DEBUG, msg_args=[e], error_code=0000, exc_info=True)
```
Comment (jruel4) [inline jf_agent/jf_jira/__init__.py:314]: Nit: Change to logger.debug

File: jf_agent/jf_jira/jira_download.py (hunk 1 above)
Comment (jruel4) [reply jf_agent/jf_jira/jira_download.py:874]: `teams = retry_for_429s(jira_connection._get_json, 'team', base=teams_url)`

File: jf_agent/jf_jira/__init__.py (hunk 2 above)
Comment (thingscouldbeworse) [reply jf_agent/jf_jira/__init__.py]: https://github.com/Jellyfish-AI/jf_agent/pull/284/commits/1d1056a4b988b271aa79d69242f543089d93512a

File: jf_agent/jf_jira/jira_download.py (hunk 1 above)
Comment (thingscouldbeworse) [reply jf_agent/jf_jira/jira_download.py]: https://github.com/Jellyfish-AI/jf_agent/pull/284/commits/471e515142bd0dd50c10ff5c682474e3b259436b

**Review by jruel4 — APPROVED** [review]
:shipit:

**Pull request merged**